text/0011-validate-requests.md

```diff
@@ -20,16 +20,16 @@ This RFC proposes to add support for checking that a request conforms to a schem
 ## Basic example
 
 Say the schema specifies that an action `readFile` should only be used with principals of type `User` and resources of type `File`.
-This RFC proposes to change the signature of the `Request` constructor to add an optional schema argument that can be used to check  argument types when constructing a `Request`.
+This RFC proposes to add a new constructor for the `Request` type, which takes a schema argument that is used to validate fields when constructing a `Request`.
 
 Examples (using informal syntax for EUIDs):
 
 ```rust
-Request::new(Some(principal), Some(action), Some(resource), context, None); // no schema; works like before
+Request::new(Some(principal), Some(action), Some(resource), context); // original API; unchanged by this RFC
 
-Request::new(Some(User::"alice"), Some(Action::"readFile"), Some(File::"secret_file.txt"), context, Some(schema)); // ok
+Request::new_with_validation(Some(User::"alice"), Some(Action::"readFile"), Some(File::"secret_file.txt"), context, Some(schema)); // returns a Request
 
-Request::new(Some(User::"alice"), Some(Action::"readFile"), Some(Folder::"some_folder"), context, Some(schema)); // error, invalid resource
+Request::new_with_validation(Some(User::"alice"), Some(Action::"readFile"), Some(Folder::"some_folder"), context, Some(schema)); // returns an error (invalid resource)
```

Comment:
Drop the `Some(...)` part here, since the schema is required for this new constructor.